src/clr-angular/utils/drag-and-drop/providers/drag-event-listener.spec.ts

```diff
@@ -0,0 +1,344 @@
+
+                // dragmove should fire
+                emulateEventOn(moveEvent, 44, 55);
+                expectEventPropValues(testComponent.dragEvent).toBe(draggableButton, 44, 55, null);
+
+                expect(testComponent.dragStartFired).toBeTruthy();
+                expect(testComponent.dragMoveFired).toBeTruthy();
+                expect(testComponent.dragEndFired).toBeFalsy();
+            });
+
+            it("should fire consecutive dragmove events on mouse move", function() {
+                const testPositions = [[11, 22], [33, 44], [55, 66], [77, 88], [99, 0]];
+                const nbDragMoveFired = testPositions.length;
+
+                emulateEventOn(startEvent, 0, 0, draggableButton);
+                emulateEventOn(moveEvent, 0, 0);
+
+                expect(testComponent.dragStartFired).toBeTruthy();
+
+                while (testPositions.length > 0) {
+                    const testPosition = testPositions.pop();
+                    emulateEventOn(moveEvent, testPosition[0], testPosition[1]);
+                    expectEventPropValues(testComponent.dragEvent)
+                        .toBe(draggableButton, testPosition[0], testPosition[1], null);
+                }
+
+                expect(testComponent.nbDragMoveFired).toBe(nbDragMoveFired);
+                expect(testComponent.dragEndFired).toBeFalsy();
+            });
+
+            it(`${moveEvent} after multiple ${startEvent} and ${endEvent} should not fire any drag events`, function() {
+                // mousedown+mouseup means it just ended prematurely before firing dragstart
+                emulateEventOn(startEvent, 0, 0, draggableButton);
+                emulateEventOn(endEvent, 0, 0);
+
+                // emulating the above action twice
+                emulateEventOn(startEvent, 0, 0, draggableButton);
+                emulateEventOn(endEvent, 0, 0);
+
+                emulateEventOn(moveEvent, 0, 0);
+
+                expect(testComponent.dragStartFired).toBeFalsy();
+                expect(testComponent.dragMoveFired).toBeFalsy();
+                expect(testComponent.dragEndFired).toBeFalsy();
+            });
+
+            it(`${startEvent} and ${moveEvent} after multiple ${startEvent} and ${
+                   endEvent} should fire dragmove events properly`,
+               function() {
+                   const testPositions = [[11, 22], [33, 44], [55, 66], [77, 88], [99, 0]];
+                   const nbDragMoveFired = testPositions.length;
+
+                   // mousedown+mouseup means it just ended prematurely
+                   emulateEventOn(startEvent, 0, 0, draggableButton);
+                   emulateEventOn(endEvent, 0, 0);
+
+                   // emulating twice
+                   emulateEventOn(startEvent, 0, 0, draggableButton);
+                   emulateEventOn(endEvent, 0, 0);
+
+                   // mousedown+mousemove shouldn't fire dragstart
```

Comment:
`mousedown + mousemove` should fire `dragStart`. no?

Reply:
You are testing it correctly. I think the comment is incorrect.

Reply:
fixed the comment message.